back/routes/experience.js

```diff
@@ -32,4 +31,71 @@ router.post('/submit', async(req, res) => {
   res.send(true).status(200);
 })
 
-module.exports = router;
\ No newline at end of file
+// -_-_-_-_-_-_-_-_ QUERIES -_-_-_-_-_-_-_
+
+// ----- Query for active feedback areas -----
+async function activeFeedbackQuery(username){
+  const query = new Parse.Query('Experience');
+  query.equalTo("username", username);
+  let experienceInfo = await query.first();
+  return experienceInfo.toJSON().activeFeedback
+}
+
+// ----- Query for active feedback areas -----
+async function allFeedbackQuery(){
+  let query = new Parse.Query('Feedback');
+  let feedbackInfo = await query.find();
+  return feedbackInfo
+}
+
+// -_-_-_-_-_-_-_-_ ENDPOINTS -_-_-_-_-_-_-_
+
+// ------ Get active feedback areas IDs -----
+router.post('/feedback/active', async(req, res) => {
+  let activeFeedback = await activeFeedbackQuery(req.body.username);
+  res.status(200).send(activeFeedback)
+})
+
+// ----- Get inactive feedback areas IDs ------
+router.post('/feedback/inactive', async(req, res) => {
+  // Find active areas
+  let activeFeedback = await activeFeedbackQuery(req.body.username);
+  // Find all feedback areas
+  let allFeedback = await allFeedbackQuery();
+  // Discard active from all to get inactive
+  let inactiveFeedback = allFeedback.filter((feedback) => !activeFeedback.includes(feedback.toJSON().objectId))
+  // Return only the id of the feedback
+  let inactiveFeedbackIDs = inactiveFeedback.map((feedback) => feedback.toJSON().objectId)
+  res.status(200).send(inactiveFeedbackIDs);
+})
+
+// ----- Get feedback information of a given array -----
+router.post('/feedback/info', async(req, res) => {
+  let allFeedback = await allFeedbackQuery();
+  // let feedbackInfo = allFeedback.filter(())
+  let feedbackInfo = req.body.IDs.map((id) => 
+    allFeedback.find((feedback) =>
+      feedback.toJSON().objectId === id)
+  )
+  let feedbackInfoJSON = feedbackInfo.map((feedback) => feedback.toJSON())
+  res.status(200).send(feedbackInfoJSON)
+})
+
+// ----- Submit feedback preferences to db -----
+router.post('/feedback/update', async(req, res) => {
+  const findQuery = new Parse.Query('Experience');
+  findQuery.equalTo("username", req.body.username);
+  let currentExperience = await findQuery.first();
+  let objectId = currentExperience.toJSON().objectId;
```

Comment:
hint: Wherever you need to call toJSON(), there's probably an easier way with Parse.

Reply:
Ok! I created an issue for that, but will prioritize other issues for now